operator/cluster/upgrade.go

```diff
@@ -100,9 +102,15 @@ func AddUpgrade(clientset *kubernetes.Clientset, restclient *rest.RESTClient, up
 
 	publishMinorUpgradeStartedEvent(&currentTask, &cl, namespace)
 
-	// start the upgrade
-	ProcessNextUpgradeItem(clientset, restclient, cl, currentTask.Spec.Name, namespace)
-
+	// if autofail is enabled, then patch the replicas, followed by the primary, sequentially.
+	// this process is started by calling ProcessNextUpgradeItem to start the upgrade
+	// otherwise if autofail is disabled, upgrade the primary and all replicas at the same time
+	// by calling UpgradeAutofailDisabled
+	if util.IsAutofailEnabled(&cl) {
+		ProcessNextUpgradeItem(clientset, restclient, cl, currentTask.Name, namespace)
+	} else {
+		UpgradeAutofailDisabled(clientset, restclient, cl, currentTask.Name, namespace)
+	}
```

Comment:
I would suggest some function renames to help with clarity, i.e.:

- `UpgradeWithAutofailEnabled` (I'm cool with "breaking the API" on this one)
- ~UpgradeWithAutofailDisabled`

Reply:
Function `UpgradeAutofailDisabled` has been renamed to `UpgradeWithAutofailDisabled`, and function `UpgradeWithAutofailEnabled` has now been created (and `UpgradeWithAutofailEnabled` is also now called instead of `ProcessNextUpgradeItem`).